_This project was created by Oluwaferanmi Alalade_
This is the progress as at -9/09/2022

In [145]:
import pandas as pd
from datetime import datetime, timedelta

In [146]:
df = pd.read_csv(r"C:\Users\Oluwaferanmi\Documents\settlements_data_JS.csv")

In [147]:
df.head()

Order ID Activity Date Bakery Item  Quantity Dee's Location Activity Type  \
0    344545    21/08/2020         CCC       170            1JS         Order   
1   4384815    20/08/2020         CCC        80            2WS         Order   
2     24084    23/08/2020        AAPL       250            2WS      Shipment   
3   3092445    21/08/2020        BBRD        50            2WS         Order   
4   2373263    21/08/2020        BBRD       120            1JS         Order   

   Inventory Date  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN

In [148]:
df["Inventory Date"].value_counts()

Series([], Name: Inventory Date, dtype: int64)

In [149]:
def InventoryDate(data):
    Invent_Date = []
    for index, record in data.iterrows():
        if record["Activity Type"] == "Order":
            date = datetime.strptime(record["Activity Date"], "%d/%m/%Y")
            record["Inventory Date"] = date + timedelta(days=2)
        elif record["Activity Type"] == "Shipment":
            if record["Dee's Location"] == "1JS" or record["Dee's Location"] == "2WS":
                date = datetime.strptime(record["Activity Date"], "%d/%m/%Y")
                record["Inventory Date"] = date + timedelta(days=1)
            elif record["Dee's Location"] == "3PUFF":
                date = datetime.strptime(record["Activity Date"], "%d/%m/%Y")
                record["Inventory Date"] = date + timedelta(days=2)
        else:
            continue

        Invent_Date.append(record["Inventory Date"])
    return Invent_Date
    

In [150]:
df["Inventory Date"] = InventoryDate(df)

In [151]:
df.head()

Order ID Activity Date Bakery Item  Quantity Dee's Location Activity Type  \
0    344545    21/08/2020         CCC       170            1JS         Order   
1   4384815    20/08/2020         CCC        80            2WS         Order   
2     24084    23/08/2020        AAPL       250            2WS      Shipment   
3   3092445    21/08/2020        BBRD        50            2WS         Order   
4   2373263    21/08/2020        BBRD       120            1JS         Order   

  Inventory Date  
0     2020-08-23  
1     2020-08-22  
2     2020-08-24  
3     2020-08-23  
4     2020-08-23

In [152]:
shipment_df = df.loc[df["Activity Type"] == "Shipment"]
order_df = df.loc[df["Activity Type"] == "Order"]

shipment_df

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
2       24084    23/08/2020        AAPL       250            2WS   
8     4755979    21/08/2020        AAPL       150            2WS   
19     199073    21/08/2020        AAPL       200          3PUFF   
21     612627    23/08/2020        AAPL       300            1JS   
22    2793436    21/08/2020        BBRD       900            1JS   
29    3836973    21/08/2020        BBRD       700            2WS   
38    4626866    22/08/2020         CCC       800          3PUFF   
41    1088540    23/08/2020        BBRD       800            1JS   
52    1473291    23/08/2020         CCC       450            1JS   
69    1901900    23/08/2020         CCC       550            2WS   
79    2125482    23/08/2020         CCC       600          3PUFF   
82     905782    22/08/2020        AAPL       350            2WS   
90    2899092    21/08/2020         CCC       450            2WS   
91    2683759    22/08/2020        BBRD       400          3PUFF   
97    4618025    22/08/2020         CCC       450            1JS   
98    2575976    23/08/2020        BBRD       400          3PUFF   
102   4546348    21/08/2020         CCC       450            1JS   
118   1452264    22/08/2020         CCC       350            2WS   
120   3059395    23/08/2020        BBRD       800            2WS   
127   2724679    21/08/2020        BBRD       400          3PUFF   
136    368117    22/08/2020        BBRD       600            2WS   
160    580731    22/08/2020        AAPL       200          3PUFF   
164   2074756    22/08/2020        BBRD      1000            1JS   
177   4928620    22/08/2020        AAPL       150            1JS   
183    163422    21/08/2020         CCC       700          3PUFF   
187   4499539    23/08/2020        AAPL       150          3PUFF   
197    131584    21/08/2020        AAPL       350            1JS   

    Activity Type Inventory Date  
2        Shipment     2020-08-24  
8        Shipment     2020-08-22  
19       Shipment     2020-08-23  
21       Shipment     2020-08-24  
22       Shipment     2020-08-22  
29       Shipment     2020-08-22  
38       Shipment     2020-08-24  
41       Shipment     2020-08-24  
52       Shipment     2020-08-24  
69       Shipment     2020-08-24  
79       Shipment     2020-08-25  
82       Shipment     2020-08-23  
90       Shipment     2020-08-22  
91       Shipment     2020-08-24  
97       Shipment     2020-08-23  
98       Shipment     2020-08-25  
102      Shipment     2020-08-22  
118      Shipment     2020-08-23  
120      Shipment     2020-08-24  
127      Shipment     2020-08-23  
136      Shipment     2020-08-23  
160      Shipment     2020-08-24  
164      Shipment     2020-08-23  
177      Shipment     2020-08-23  
183      Shipment     2020-08-23  
187      Shipment     2020-08-25  
197      Shipment     2020-08-22

## Shipment

In [153]:
shipment_df.groupby(shipment_df["Bakery Item"])[["Quantity"]].sum()

Quantity
Bakery Item          
AAPL             2100
BBRD             6000
CCC              4800

From the above, it is seen that in the four day period, 2100 Applepies, 6000 banana bread and 4800 Chocolate chip cookies wereshipped

In [154]:
day1_ship = shipment_df.loc[shipment_df["Activity Date"] == "20/08/2020"]

In [155]:
day1_ship.head()

Empty DataFrame
Columns: [Order ID, Activity Date, Bakery Item, Quantity, Dee's Location, Activity Type, Inventory Date]
Index: []

In [156]:
day1_ship.groupby(day1_ship["Bakery Item"]).sum()[["Quantity"]]

Empty DataFrame
Columns: [Quantity]
Index: []

In [157]:
day2_ship = shipment_df.loc[shipment_df["Activity Date"] == "21/08/2020"]
day2_ship

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
8     4755979    21/08/2020        AAPL       150            2WS   
19     199073    21/08/2020        AAPL       200          3PUFF   
22    2793436    21/08/2020        BBRD       900            1JS   
29    3836973    21/08/2020        BBRD       700            2WS   
90    2899092    21/08/2020         CCC       450            2WS   
102   4546348    21/08/2020         CCC       450            1JS   
127   2724679    21/08/2020        BBRD       400          3PUFF   
183    163422    21/08/2020         CCC       700          3PUFF   
197    131584    21/08/2020        AAPL       350            1JS   

    Activity Type Inventory Date  
8        Shipment     2020-08-22  
19       Shipment     2020-08-23  
22       Shipment     2020-08-22  
29       Shipment     2020-08-22  
90       Shipment     2020-08-22  
102      Shipment     2020-08-22  
127      Shipment     2020-08-23  
183      Shipment     2020-08-23  
197      Shipment     2020-08-22

In [158]:
day2_ship.groupby("Bakery Item").sum()[["Quantity"]]

Quantity
Bakery Item          
AAPL              700
BBRD             2000
CCC              1600

In [159]:
day3_ship = shipment_df.loc[shipment_df["Activity Date"] == "22/08/2020"]
day3_ship

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
38    4626866    22/08/2020         CCC       800          3PUFF   
82     905782    22/08/2020        AAPL       350            2WS   
91    2683759    22/08/2020        BBRD       400          3PUFF   
97    4618025    22/08/2020         CCC       450            1JS   
118   1452264    22/08/2020         CCC       350            2WS   
136    368117    22/08/2020        BBRD       600            2WS   
160    580731    22/08/2020        AAPL       200          3PUFF   
164   2074756    22/08/2020        BBRD      1000            1JS   
177   4928620    22/08/2020        AAPL       150            1JS   

    Activity Type Inventory Date  
38       Shipment     2020-08-24  
82       Shipment     2020-08-23  
91       Shipment     2020-08-24  
97       Shipment     2020-08-23  
118      Shipment     2020-08-23  
136      Shipment     2020-08-23  
160      Shipment     2020-08-24  
164      Shipment     2020-08-23  
177      Shipment     2020-08-23

In [160]:
day3_ship.groupby(day3_ship["Bakery Item"]).sum()

Order ID  Quantity
Bakery Item                    
AAPL          6415133       700
BBRD          5126632      2000
CCC          10697155      1600

In [161]:
day4_ship = shipment_df.loc[shipment_df["Activity Date"] == "23/08/2020"]
day4_ship

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
2       24084    23/08/2020        AAPL       250            2WS   
21     612627    23/08/2020        AAPL       300            1JS   
41    1088540    23/08/2020        BBRD       800            1JS   
52    1473291    23/08/2020         CCC       450            1JS   
69    1901900    23/08/2020         CCC       550            2WS   
79    2125482    23/08/2020         CCC       600          3PUFF   
98    2575976    23/08/2020        BBRD       400          3PUFF   
120   3059395    23/08/2020        BBRD       800            2WS   
187   4499539    23/08/2020        AAPL       150          3PUFF   

    Activity Type Inventory Date  
2        Shipment     2020-08-24  
21       Shipment     2020-08-24  
41       Shipment     2020-08-24  
52       Shipment     2020-08-24  
69       Shipment     2020-08-24  
79       Shipment     2020-08-25  
98       Shipment     2020-08-25  
120      Shipment     2020-08-24  
187      Shipment     2020-08-25

In [162]:
day4_ship.groupby(by="Bakery Item").sum()

Order ID  Quantity
Bakery Item                    
AAPL          5136250       700
BBRD          6723911      2000
CCC           5500673      1600

## Order 

In [163]:
order_df.groupby("Bakery Item").sum()[["Quantity"]]

Quantity
Bakery Item          
AAPL             3960
BBRD             6420
CCC              5710

3960 apple pies were ordered, 6420 banana bread and 5710 chocolate chip cookies 

### First day

In [164]:
# day1_ord = order_df.loc[order_df["Activity Date"] == "20/08/2020"]
# day1_ord.head()

In [165]:
# day1_ord.groupby("Bakery Item").sum()[["Quantity"]]

In [166]:
# day2_ord = order_df.loc[order_df["Activity Date"] == "21/08/2020"]
# day2_ord.head()

In [167]:
# day2_ord.groupby("Bakery Item").sum()[["Quantity"]]

In [168]:
# day3_ord = order_df.loc[order_df["Activity Date"] == "22/08/2020"]
# day3_ord.head()

In [169]:
# day3_ord.groupby("Bakery Item").sum()[["Quantity"]]

In [170]:
# day4_ord = order_df.loc[order_df["Activity Date"] == "23/08/2020"]
# day4_ord.head()

In [171]:
# base_aapl = 500
# base_bbrd = 500
# base_ccc = 2120

## Shipping to different locations
### Finding the quantity of each product shipped to the different locations on the 22nd, 23rd and 24th

#### The inventory dates

In [172]:
shipment_df["Inventory Date"].value_counts()

2020-08-24    9
2020-08-23    9
2020-08-22    6
2020-08-25    3
Name: Inventory Date, dtype: int64

In [173]:
#Grouping the different shipments by the inventory dates
d = shipment_df.groupby("Inventory Date")

In [174]:
#Getting the information on the first inventory date which is the 22nd of august, 2020

first_date = d.get_group("2020-08-22")
first_date

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
8     4755979    21/08/2020        AAPL       150            2WS   
22    2793436    21/08/2020        BBRD       900            1JS   
29    3836973    21/08/2020        BBRD       700            2WS   
90    2899092    21/08/2020         CCC       450            2WS   
102   4546348    21/08/2020         CCC       450            1JS   
197    131584    21/08/2020        AAPL       350            1JS   

    Activity Type Inventory Date  
8        Shipment     2020-08-22  
22       Shipment     2020-08-22  
29       Shipment     2020-08-22  
90       Shipment     2020-08-22  
102      Shipment     2020-08-22  
197      Shipment     2020-08-22

Analyzing information on the quantity of products shipped to the different locations on the first inventory date
 - js_first being the dataframe which shows the quantity of each bakery items shipped to 1JS outlet on the first inventory date
 - ws_first being the dataframe which shows the quantity of each bakery items shipped to 2WS outlet on the first inventory date

In [175]:
location_first = first_date.groupby("Dee's Location")#Grouping by the different outlet locations for the first inv date
js_first_ship = location_first.get_group("1JS")[["Bakery Item","Quantity"]]
js_first_ship = js_first_ship.groupby("Bakery Item").sum()["Quantity"]

ws_first_ship = location_first.get_group("2WS")[["Bakery Item","Quantity"]]
ws_first_ship = ws_first_ship.groupby("Bakery Item").sum()["Quantity"]


In [176]:
js_first_ship

Bakery Item
AAPL    350
BBRD    900
CCC     450
Name: Quantity, dtype: int64

In [177]:
ws_first_ship

Bakery Item
AAPL    150
BBRD    700
CCC     450
Name: Quantity, dtype: int64

In [178]:
second_date = d.get_group("2020-08-23")
second_date

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
19     199073    21/08/2020        AAPL       200          3PUFF   
82     905782    22/08/2020        AAPL       350            2WS   
97    4618025    22/08/2020         CCC       450            1JS   
118   1452264    22/08/2020         CCC       350            2WS   
127   2724679    21/08/2020        BBRD       400          3PUFF   
136    368117    22/08/2020        BBRD       600            2WS   
164   2074756    22/08/2020        BBRD      1000            1JS   
177   4928620    22/08/2020        AAPL       150            1JS   
183    163422    21/08/2020         CCC       700          3PUFF   

    Activity Type Inventory Date  
19       Shipment     2020-08-23  
82       Shipment     2020-08-23  
97       Shipment     2020-08-23  
118      Shipment     2020-08-23  
127      Shipment     2020-08-23  
136      Shipment     2020-08-23  
164      Shipment     2020-08-23  
177      Shipment     2020-08-23  
183      Shipment     2020-08-23

 Second date - 23rd of august 2020
 - js_sec being the dataframe which shows the quantity of each bakery items shipped to 1JS outlet on the second inventory date
 - ws_sec being the dataframe which shows the quantity of each bakery items shipped to 2WS outlet on the second inventory date

In [179]:
location_sec = second_date.groupby("Dee's Location")
js_sec_ship = location_sec.get_group("1JS")[["Bakery Item","Quantity"]]
js_sec_ship = js_sec_ship.groupby("Bakery Item").sum()["Quantity"]

ws_sec_ship = location_sec.get_group("2WS")[["Bakery Item","Quantity"]]
ws_sec_ship = ws_sec_ship.groupby("Bakery Item").sum()["Quantity"]

pf_sec_ship = location_sec.get_group("3PUFF")[["Bakery Item","Quantity"]]
pf_sec_ship = pf_sec_ship.groupby("Bakery Item").sum()["Quantity"]

In [180]:
js_sec_ship

Bakery Item
AAPL     150
BBRD    1000
CCC      450
Name: Quantity, dtype: int64

In [181]:
ws_sec_ship

Bakery Item
AAPL    350
BBRD    600
CCC     350
Name: Quantity, dtype: int64

In [182]:
pf_sec_ship

Bakery Item
AAPL    200
BBRD    400
CCC     700
Name: Quantity, dtype: int64

Third date - 24th August 2020

In [183]:
third_date = d.get_group("2020-08-24")
third_date

Order ID Activity Date Bakery Item  Quantity Dee's Location  \
2       24084    23/08/2020        AAPL       250            2WS   
21     612627    23/08/2020        AAPL       300            1JS   
38    4626866    22/08/2020         CCC       800          3PUFF   
41    1088540    23/08/2020        BBRD       800            1JS   
52    1473291    23/08/2020         CCC       450            1JS   
69    1901900    23/08/2020         CCC       550            2WS   
91    2683759    22/08/2020        BBRD       400          3PUFF   
120   3059395    23/08/2020        BBRD       800            2WS   
160    580731    22/08/2020        AAPL       200          3PUFF   

    Activity Type Inventory Date  
2        Shipment     2020-08-24  
21       Shipment     2020-08-24  
38       Shipment     2020-08-24  
41       Shipment     2020-08-24  
52       Shipment     2020-08-24  
69       Shipment     2020-08-24  
91       Shipment     2020-08-24  
120      Shipment     2020-08-24  
160      Shipment     2020-08-24

In [184]:
location_third = third_date.groupby("Dee's Location")
js_third_ship = location_third.get_group("1JS")[["Bakery Item","Quantity"]]
js_third_ship = js_third_ship.groupby("Bakery Item").sum()["Quantity"]

ws_third_ship = location_third.get_group("2WS")[["Bakery Item","Quantity"]]
ws_third_ship = ws_third_ship.groupby("Bakery Item").sum()["Quantity"]

pf_third_ship = location_third.get_group("3PUFF")[["Bakery Item","Quantity"]]
pf_third_ship = pf_third_ship.groupby("Bakery Item").sum()["Quantity"]

In [185]:
js_third_ship

Bakery Item
AAPL    300
BBRD    800
CCC     450
Name: Quantity, dtype: int64

In [186]:
ws_third_ship

Bakery Item
AAPL    250
BBRD    800
CCC     550
Name: Quantity, dtype: int64

In [187]:
pf_third_ship

Bakery Item
AAPL    200
BBRD    400
CCC     800
Name: Quantity, dtype: int64

## Ordersfrom different locations
### Finding the quantity of each product ordered from the different locations on the 22nd, 23rd and 24th

In [188]:
e = order_df.groupby("Inventory Date") #Grouping the different orders by the inventory dates

### First Inventory Date

In [189]:
first_ord = e.get_group("2020-08-22")

In [190]:
location_first = first_ord.groupby("Dee's Location")
js_first_ord = location_first.get_group("1JS")


In [191]:
js_first_ord = js_first_ord.groupby("Bakery Item").sum()["Quantity"]

In [192]:
js_first_ord

Bakery Item
AAPL    680
BBRD    520
CCC     440
Name: Quantity, dtype: int64

In [193]:
ws_first_ord = location_first.get_group("2WS")
ws_first_ord = ws_first_ord.groupby("Bakery Item").sum()["Quantity"]
ws_first_ord

Bakery Item
AAPL    320
BBRD    530
CCC     500
Name: Quantity, dtype: int64

In [194]:
pf_first_ord = location_first.get_group("3PUFF")
pf_first_ord = pf_first_ord.groupby("Bakery Item").sum()["Quantity"]
pf_first_ord

Bakery Item
AAPL    420
BBRD    250
CCC     290
Name: Quantity, dtype: int64

### Second Inventory Date

In [195]:
sec_ord = e.get_group("2020-08-23")

In [196]:
location_sec_ord = sec_ord.groupby("Dee's Location")
js_sec_ord = location_sec_ord.get_group("1JS")

In [197]:
js_sec_ord = js_sec_ord.groupby("Bakery Item").sum()["Quantity"]
js_sec_ord

Bakery Item
AAPL     520
BBRD    1190
CCC     1710
Name: Quantity, dtype: int64

In [198]:
js_sec_ord

Bakery Item
AAPL     520
BBRD    1190
CCC     1710
Name: Quantity, dtype: int64

In [199]:
ws_sec_ord = location_sec_ord.get_group("2WS")
ws_sec_ord = ws_sec_ord.groupby("Bakery Item").sum()["Quantity"]
ws_sec_ord

Bakery Item
AAPL    330
BBRD    660
CCC     230
Name: Quantity, dtype: int64

In [200]:
pf_sec_ord = location_sec_ord.get_group("3PUFF")
pf_sec_ord = pf_sec_ord.groupby("Bakery Item").sum()["Quantity"]
pf_sec_ord

Bakery Item
AAPL    410
BBRD    600
CCC     420
Name: Quantity, dtype: int64

### Third Inventory Date

In [201]:
third_ord = e.get_group("2020-08-24")

In [202]:
location_third_ord = third_ord.groupby("Dee's Location")

js_third_ord = location_third_ord.get_group("1JS")
js_third_ord = js_third_ord.groupby("Bakery Item").sum()["Quantity"]

js_third_ord

Bakery Item
AAPL     770
BBRD    1310
CCC     1090
Name: Quantity, dtype: int64

In [203]:
ws_third_ord = location_third_ord.get_group("2WS")
ws_third_ord = ws_third_ord.groupby("Bakery Item").sum()["Quantity"]
ws_third_ord

Bakery Item
AAPL    340
BBRD    700
CCC     590
Name: Quantity, dtype: int64

In [204]:
pf_third_ord = location_third_ord.get_group("3PUFF")
pf_third_ord = pf_third_ord.groupby("Bakery Item").sum()["Quantity"]
pf_third_ord

Bakery Item
AAPL    170
BBRD    660
CCC     440
Name: Quantity, dtype: int64

## To calculate the quantity of each product in the inventory of each outlet and the end of each day
The earliest shipment to any of the outlet stores landed on the 22nd of August, before then it is assumed that all the outlets had 500 of each product in their inventory, and it is stated that AAPL shipment was on hold due to logistic issues which means the quantity of Apple pies in each of the outlets remained 500 throughout the period

In [205]:
base_aapl = 500
base_bbrd = 500
base_ccc = 500

In [206]:
js_first = {}

for index,(js1_ship, js1_ord) in enumerate(zip(js_first_ship.values, js_first_ord.values)):
    if index == 0:
        js_first["AAPL"] = [js1_ship, js1_ord]
    elif index == 1:
        js_first["BBRD"] = [js1_ship, js1_ord]
    elif index == 2:
        js_first["CCC"] = [js1_ship, js1_ord]

In [207]:
js_first

{'AAPL': [350, 680], 'BBRD': [900, 520], 'CCC': [450, 440]}

In [208]:
js_second = {}

for index,(js2_ship, js2_ord) in enumerate(zip(js_sec_ship.values, js_sec_ord.values)):
    if index == 0:
        js_second["AAPL"] = [js2_ship, js2_ord]
    elif index == 1:
        js_second["BBRD"] = [js2_ship, js2_ord]
    elif index == 2:
        js_second["CCC"] = [js2_ship, js2_ord]

In [209]:
js_second

{'AAPL': [150, 520], 'BBRD': [1000, 1190], 'CCC': [450, 1710]}

In [210]:
js_third = {}

for index,(js3_ship, js3_ord) in enumerate(zip(js_third_ship.values, js_third_ord.values)):
    if index == 0:
        js_third["AAPL"] = [js3_ship, js3_ord]
    elif index == 1:
        js_third["BBRD"] = [js3_ship, js3_ord]
    elif index == 2:
        js_third["CCC"] = [js3_ship, js3_ord]

In [211]:
js_third

{'AAPL': [300, 770], 'BBRD': [800, 1310], 'CCC': [450, 1090]}

In [212]:
js = {}

js["First Day"] = js_first
js["Second Day"] = js_second
js["Third Day"] = js_third

In [213]:
js

{'First Day': {'AAPL': [350, 680], 'BBRD': [900, 520], 'CCC': [450, 440]},
 'Second Day': {'AAPL': [150, 520], 'BBRD': [1000, 1190], 'CCC': [450, 1710]},
 'Third Day': {'AAPL': [300, 770], 'BBRD': [800, 1310], 'CCC': [450, 1090]}}

In [214]:
js.get("First Day").get("AAPL")

[350, 680]

In [215]:
def ord_ship_bal(data):
    balance = int(data[0]) - int(data[1])
    return balance

In [216]:
base_aapl = 500
base_bbrd = 500
base_ccc = 500
invent_day = {}
first_day = {}
second_day = {}
third_day = {}

for index, day in enumerate(js.values()):
    print(day)
    
    if index == 0:
        first_day["AAPL"] = base_aapl - day.get("AAPL")[1]
        first_day["BBRD"] = base_bbrd + ord_ship_bal(day.get("BBRD"))
        first_day["CCC"] = base_ccc + ord_ship_bal(day.get("CCC"))
        
        base_aapl = first_day.get("AAPL")
        base_bbrd = first_day["BBRD"]
        base_ccc = first_day["CCC"]
    
    if index == 1:
        second_day["AAPL"] = base_aapl - day.get("AAPL")[1]
        second_day["BBRD"] = base_bbrd + ord_ship_bal(day.get("BBRD"))
        second_day["CCC"] = base_ccc + ord_ship_bal(day.get("CCC"))
        
        base_aapl = second_day.get("AAPL")
        base_bbrd = second_day["BBRD"]
        base_ccc = second_day["CCC"]
    
    if index == 2:
        third_day["AAPL"] = base_aapl - day.get("AAPL")[1]
        third_day["BBRD"] = base_bbrd + ord_ship_bal(day.get("BBRD"))
        third_day["CCC"] = base_ccc + ord_ship_bal(day.get("CCC"))
    

{'AAPL': [350, 680], 'BBRD': [900, 520], 'CCC': [450, 440]}
{'AAPL': [150, 520], 'BBRD': [1000, 1190], 'CCC': [450, 1710]}
{'AAPL': [300, 770], 'BBRD': [800, 1310], 'CCC': [450, 1090]}


In [217]:
first_day

{'AAPL': -180, 'BBRD': 880, 'CCC': 510}

In [218]:
second_day

{'AAPL': -700, 'BBRD': 690, 'CCC': -750}

In [219]:
third_day

{'AAPL': -1470, 'BBRD': 180, 'CCC': -1390}